addons/web/static/src/js/view_list.js

```diff
@@ -2305,13 +2305,9 @@ instance.web.list.Binary = instance.web.list.Column.extend({
         var text = _t("Download");
         var value = row_data[this.id].value;
         var download_url;
-        if (value && value.substr(0, 10).indexOf(' ') == -1) {
-            download_url = "data:application/octet-stream;base64," + value;
```

Comment:
Except that the content is already fetched when you load the o2m (check the `read` call) so you already have it. You can fetch the filename using `row_data` to fill the download attribute.